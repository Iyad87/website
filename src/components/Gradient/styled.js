import styled from 'styled-components';

export const BaseGradient = styled.div`
    position: absolute;
    bottom: 0;
    left: -7.5%;
    height: ${props => props.height};
    width: 115%;
    background-image: linear-gradient(
      -180deg,
      rgba(255, 255, 255, 0.5) 0%,
      #f3f5fa 52%
    );
    z-index: -1;
  }
`;
