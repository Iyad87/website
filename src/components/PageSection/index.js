import Container from '../Container';
import { spacing } from 'styles/spacing';

const PageSection = Container.withComponent('section').extend`
  margin-bottom: ${spacing(10)};
  margin-top: ${spacing(10)};
`;

export default PageSection;
