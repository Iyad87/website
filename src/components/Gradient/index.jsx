import React from 'react';
import PropTypes from 'prop-types';
import { BaseGradient } from './styled';

const Gradient = ({ height, ...props }) => (
  <BaseGradient height={height} {...props} />
);

Gradient.propTypes = {
  height: PropTypes.string,
};

Gradient.defaultProps = {
  height: '50%',
};

export default Gradient;
