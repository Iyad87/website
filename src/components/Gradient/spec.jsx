import React from 'react';
import { shallow } from 'enzyme';
import Gradient from './index';

test('snapshot', () => {
  const wrapper = shallow(<Gradient />);

  expect(wrapper).toMatchSnapshot();
});

test('applies height property', () => {
  const wrapper = shallow(<Gradient height="100px" />);

  expect(wrapper.prop('height')).toEqual('100px');
});
