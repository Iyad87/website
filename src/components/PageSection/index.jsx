import React from 'react';
import PropTypes from 'prop-types';

import Gradient from 'components/Gradient';
import { Section } from './styled';

const PageSection = ({ children, gradient, gradientHeight, ...props }) => (
  <Section gradient={gradient} {...props}>
    {children}
    {gradient && <Gradient height={gradientHeight} />}
  </Section>
);

PageSection.propTypes = {
  children: PropTypes.node,
  gradient: PropTypes.bool,
  gradientHeight: PropTypes.string,
};

PageSection.defaultProps = {
  children: [],
  gradient: false,
  gradientHeight: '50%',
};

export default PageSection;
