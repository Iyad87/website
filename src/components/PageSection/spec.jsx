import React from 'react';
import { shallow } from 'enzyme';
import PageSection from './index';
import Gradient from 'components/Gradient';

test('snapshot', () => {
  const wrapper = shallow(<PageSection />);

  expect(wrapper).toMatchSnapshot();
});

test('snapshot with gradient', () => {
  const wrapper = shallow(<PageSection gradient>Content</PageSection>);

  expect(wrapper).toMatchSnapshot();
});

test('renders a gradient when provided', () => {
  const wrapper = shallow(
    <PageSection gradient>
      <div>Content</div>
    </PageSection>,
  );

  const gradient = wrapper
    .find('Container')
    .children()
    .last();

  expect(gradient.is(Gradient)).toBe(true);
});
