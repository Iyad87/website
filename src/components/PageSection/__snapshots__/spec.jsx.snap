// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<Container
  gradient={false}
/>
`;

exports[`snapshot with gradient 1`] = `
<Container
  gradient={true}
>
  Content
  <Gradient
    height="50%"
  />
</Container>
`;
